### PR TITLE
replaced custom retry function with tenacity retrying library.

### PR DESCRIPTION
**Pull request by Pranay144**

### CODE REVIEW

File: bugbug/utils.py [hunk 1]
```diff
@@ -199,16 +199,12 @@ def download_and_load_model(model_name):
     return get_model_class(model_name).load(path)
 
 
-def retry(operation, retries=5, wait_between_retries=30):
+def retrying(operation):
```
Comment (marco-c) [inline bugbug/utils.py]: We should completely remove this function and just use tenacity where needed

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 repos:
```
Comment (marco-c) [inline .pre-commit-config.yaml]: The changes in this file are unrelated to the PR, please revert them

**Comment by marco-c** [comment]
What you proposed in https://github.com/mozilla/bugbug/issues/1234#issuecomment-572038642 looked better, why did you change it completely? :D

**Comment by Pranay144** [comment]
@marco-c I did the required changes.
please review.

File: scripts/commit_classifier.py [hunk 3]
```diff
@@ -22,13 +22,13 @@
 from libmozdata import vcs_map
 from libmozdata.phabricator import PhabricatorAPI
 from scipy.stats import spearmanr
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_fixed
```
Comment (marco-c) [inline scripts/commit_classifier.py]: Nit: `import tenacity` and then in the code use `tenacity.retry`, etc.

File: scripts/commit_classifier.py [hunk 4]
```diff
@@ -194,7 +194,10 @@ def clone_git_repo(self, repo_url, repo_dir, rev="master"):
             retry(
                 lambda: subprocess.run(
                     ["git", "clone", "--quiet", repo_url, repo_dir], check=True
-                )
+                ),
+                retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/commit_classifier.py]: This is not needed, we will retry on all exceptions anyway

File: scripts/commit_classifier.py [hunk 5]
```diff
@@ -203,13 +206,19 @@ def clone_git_repo(self, repo_url, repo_dir, rev="master"):
                 cwd=repo_dir,
                 capture_output=True,
                 check=True,
-            )
+            ),
+            retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/commit_classifier.py]: Same here

File: scripts/commit_classifier.py [hunk 6]
```diff
@@ -203,13 +206,19 @@ def clone_git_repo(self, repo_url, repo_dir, rev="master"):
...
+            wait=wait_fixed(30),
+            stop=stop_after_attempt(5),
         )
 
         retry(
             lambda: subprocess.run(
                 ["git", "checkout", rev], cwd=repo_dir, capture_output=True, check=True
-            )
+            ),
+            retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/commit_classifier.py]: Same here

File: scripts/microannotate_generator.py [hunk 7]
```diff
@@ -6,9 +6,10 @@
 from logging import INFO, basicConfig, getLogger
 
 from microannotate import generator
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_fixed
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: Nit: here too, import tenacity

File: scripts/microannotate_generator.py [hunk 8]
```diff
@@ -64,7 +65,10 @@ def generate(self):
         retry(
             lambda: subprocess.run(
                 ["git", "config", "--global", "http.postBuffer", "12M"], check=True
-            )
+            ),
+            retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: Same here, remove this argument

File: scripts/microannotate_generator.py [hunk 9]
```diff
@@ -81,7 +85,12 @@ def generate(self):
                 remove_comments=self.remove_comments,
             )
 
-            retry(lambda: subprocess.run(push_args, cwd=self.git_repo_path, check=True))
+            retry(
+                lambda: subprocess.run(push_args, cwd=self.git_repo_path, check=True),
+                retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: And here

File: scripts/microannotate_generator.py [hunk 10]
```diff
@@ -97,7 +106,10 @@ def clone_git_repo(self):
             lambda: subprocess.run(
                 ["git", "clone", "--quiet", self.repo_url, self.git_repo_path],
                 check=True,
-            )
+            ),
+            retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: And here

File: scripts/microannotate_generator.py [hunk 11]
```diff
@@ -107,7 +119,10 @@ def clone_git_repo(self):
                     cwd=self.git_repo_path,
                     capture_output=True,
                     check=True,
-                )
+                ),
+                retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/microannotate_generator.py]: And here

File: scripts/regressor_finder.py [hunk 12]
```diff
@@ -18,6 +18,7 @@
 from libmozdata import vcs_map
 from microannotate import utils as microannotate_utils
 from pydriller import GitRepository
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_fixed
```
Comment (marco-c) [inline scripts/regressor_finder.py]: Same here, import tenacity

File: scripts/regressor_finder.py [hunk 13]
```diff
@@ -86,7 +87,10 @@ def clone_git_repo(self, repo_url, repo_dir):
             retry(
                 lambda: subprocess.run(
                     ["git", "clone", "--quiet", repo_url, repo_dir], check=True
-                )
+                ),
+                retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/regressor_finder.py]: And here

File: scripts/regressor_finder.py [hunk 14]
```diff
@@ -95,7 +99,10 @@ def clone_git_repo(self, repo_url, repo_dir):
                 cwd=repo_dir,
                 capture_output=True,
                 check=True,
-            )
+            ),
+            retry=retry_if_exception_type(Exception),
```
Comment (marco-c) [inline scripts/regressor_finder.py]: And here

File: .pre-commit-config.yaml [hunk 15]
```diff
@@ -8,7 +8,7 @@ repos:
     hooks:
     -   id: isort
 -   repo: https://github.com/ambv/black
-    rev: stable
+    rev: 19.10b0
```
Comment (marco-c) [inline .pre-commit-config.yaml]: Please revert this unrelated change

Comment (Pranay144) [reply .pre-commit-config.yaml]: But the checks are failing if i don't add this file!

Comment (marco-c) [reply .pre-commit-config.yaml]: You need to remove the change and then run pre-commit to format your files.

Comment (Pranay144) [reply .pre-commit-config.yaml]: Ok...it's done.

**Comment by marco-c** [comment]
It looks like you didn't test this at all, it doesn't work :(
I'm going to revert it.

**Comment by Pranay144** [comment]
I am looking it...will try to resolve issue.
